m2c/if_statements.py

```diff
@@ -1467,7 +1467,10 @@ def get_function_text(function_info: FunctionInfo, options: Options) -> str:
                 type_decl = var.type.to_decl(var.format(fmt), fmt)
                 temp_decls.append(f"{type_decl};")
                 any_decl = True
-        for decl in sorted(temp_decls):
+
+        var_sort = (lambda d: d.split("_")[1][1:]) if fmt.descending_regs else None
```

Comment:
```suggestion
        var_sort = (lambda d: d.partition("_")[2]) if fmt.descending_regs else None
```
to include the full text after _ instead of just the second _-separated word, and include the first character of the register name in the sorting. Or was there a reason to drop it?

I guess float registers will still be sorted incorrectly, with f10 < f2.

Reply:
Your suggestion seems nice to me. There wasn't a particular reason not to include first character. Regarding f10 < f2, that shouldn't be a big issue, as this feature is mainly so that saved registers are in the "right order" for specific MWCC environments. Do you have a suggestion though if it's bothering you?

Reply:
I think theoretically the right fix would be to put some ordering information in `Var`, getting it from Register at the point of construction. But let's leave that be for the moment.

Hm, wait, does my suggestion lead to interleaving
```
var_t1
temp_t1
var_t1_2
temp_t1_2
...
```
? Then your original version with `split` is better. I'd still drop the `[1:]` from it.

Reply:
True, it would, okay.